packages/pwa/app/utils/url.js

```diff
@@ -185,3 +191,43 @@ export const removeQueryParamsFromPath = (path, keys) => {
 
     return `${pathname}${paramStr && '?'}${paramStr}`
 }
+
+/**
+ * Rebuild the path by adding/removing locale/site query params based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} configValues - object that contains values of url param config
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses/', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses/?locale=en-GB
+ *
+ */
+export const buildPathWithUrlConfigParams = (url, configValues = {}) => {
```

Comment:
will this be moved into the SDK? If so, we need to make sure the method signature (method name, arguments and return value) is easy to use.

A couple thoughts:

1. The argument `url` is vague. By definition, a URL is a string that is composed by the following parts: protocol, hostname, port, path, query and hash... The JSDOC example `/women/dresses/` is a path. Maybe we want to rename the argument to `path`.
2. The implementation references "config" a couple times, but they (`urlConfig`, `configValues`) refers to different objects. How about we rename `configValues` to `options`

Reply:
No, this util should be in the PWA because it uses the urlConfig to determine to build a new path